source/common/http/matching/inputs.h

```diff
@@ -26,17 +26,17 @@ class HttpHeadersDataInputBase : public Matcher::DataInput<HttpMatchingData> {
     const auto maybe_headers = headerMap(data);
 
     if (!maybe_headers) {
-      return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, absl::nullopt};
+      return {Matcher::DataInputGetResult::DataAvailability::NotAvailable, Matcher::InputValue()};
     }
 
     auto header_string = HeaderUtility::getAllOfHeaderAsString(maybe_headers->get(), name_, ",");
 
     if (header_string.result()) {
       return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
-              std::string(header_string.result().value())};
+              Matcher::InputValue(header_string.result().value())};
```

Comment:
should not we pass a list of values here? currently it is passing a comma separated list as a single string value?